ApplicationLibCode/UserInterface/RiuMainWindowBase.cpp

```diff
@@ -692,11 +700,15 @@ void RiuMainWindowBase::saveDockLayout()
 //--------------------------------------------------------------------------------------------------
 void RiuMainWindowBase::addDefaultEntriesToWindowsMenu()
 {
-    QMenu* showHideMenu = m_windowMenu->addMenu( "Show/Hide Windows" );
+    QMenu* dockWindowsMenu = m_windowMenu->addMenu( QIcon( ":/window-management.svg" ), "Windows" );
 
-    for ( auto dock : dockManager()->dockWidgetsMap() )
+    auto dockMap = dockManager()->dockWidgetsMap();
+    auto keys    = dockMap.keys();
+    keys.sort();
+    for ( auto& key : keys )
     {
-        showHideMenu->addAction( dock->toggleViewAction() );
+        auto dock = dockMap[key];
+        dockWindowsMenu->addAction( dock->toggleViewAction() );
```

Comment:
I was not able to toggle the visibility from the menu, only turn on window visibility.

Reply:
That is how the docking system works if you want to show icons in the menu. The menu is now basically a "Switch to (and show if hidden)" menu.

To turn off a view, use the x in the view titlebar/tab.